### PR TITLE
Stub out new properties/enum in URLSessionConfiguration

### PR DESCRIPTION
**Pull request by pushkarnk**
New properties per [the API](https://developer.apple.com/documentation/foundation/urlsessionconfiguration?changes=latest_minor) doc. Though they may be irrelevant in the Linux server context, they may make sense in the Android context.

### CODE REVIEW

**Comment by pushkarnk** [comment]
@johnno1962 The two new properties in URLSessionConfiguration `waitForConnectivity` and `multipathServiceType` are possibly irrelevant to the Linux server context. Do you see them being implemented in the near future for Swift Foundation on Android though?

**Comment by johnno1962** [comment]
Hi, this might be useful in future but for now I can’t think of a way to implement this from the NDK.

**Comment by pushkarnk** [comment]
@parkera Can you please let us know your opinion on this?

**Comment by parkera** [comment]
@salgernon could also chime in here, but -- it seems that if we have no way to reasonably implement this, we may be better off either leaving these out or marking them unavailable to warn developers that there is no implementation.

**Comment by parkera** [comment]
This one falls into the category of "leafy" API that doesn't need to be implemented with `NSUnimplemented`.

**Comment by alblue** [comment]
If we think these aren't going to be implemented any time soon, we could use `NSUnsupported()` and an availability macro to indicate that it's deprecated/unavailable on non-Darwin platforms, as we did in:

https://github.com/apple/swift-corelibs-foundation/blob/master/Foundation/Bundle.swift#L80

That way we allow those coding against it to know that these aren't available and will generate a compile error/warning, rather than relying on the behaviour and wondering why it doesn't work.

Of course if we do think this will be implemented in the near future then `NSUnimplemented` may be better.

**Comment by ianpartridge** [comment]
I agree that `NSUnsupported()` seems the best choice here.

File: Foundation/URLSession/URLSessionConfiguration.swift [hunk 1]
```diff
@@ -209,4 +209,19 @@ open class URLSessionConfiguration : NSObject, NSCopying {
      */
      open var protocolClasses: [AnyClass]?
 
+     /* A Boolean value that indicates whether the session should wait for connectivity to become available, or fail immediately */
+     open var waitsForConnectivity: Bool = false
```
Comment (alblue) [inline Foundation/URLSession/URLSessionConfiguration.swift]: These should have a warning:

`@available(*,deprecated,message:"Not available on non-Darwin platforms")`

File: Foundation/URLSession/URLSessionConfiguration.swift [hunk 2]
```diff
@@ -209,4 +209,19 @@ open class URLSessionConfiguration : NSObject, NSCopying {
...
+
+     /* A service type that specifies the Multipath TCP connection policy for transmitting data over Wi-Fi and cellular interfaces*/
+     open var multipathServiceType: URLSessionConfiguration.MultipathServiceType = .none
+
+}
+
```
Comment (alblue) [inline Foundation/URLSession/URLSessionConfiguration.swift]: This should have a warning:

`@available(*,deprecated,message:"Not available on non-Darwin platforms")`

Comment (parkera) [reply Foundation/URLSession/URLSessionConfiguration.swift]: We can also use `unavailable` to make it an error to attempt to use it.

**Comment by pushkarnk** [comment]
Thanks @parkera @alblue and @ianpartridge for reviewing! I agree with all the points made above and have made changes suitably.

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Pull request merged**